pkg/security/module/msg_sender.go

```diff
@@ -86,7 +86,8 @@ func NewDirectMsgSender(stopper startstop.Stopper) (*DirectMsgSender, error) {
 
 	// we set the hostname to the empty string to take advantage of the out of the box message hostname
 	// resolution
-	reporter, err := reporter.NewCWSReporter("", stopper, endpoints, destinationsCtx)
+	// TODO That nil is exceptionally bad!!!
```

Comment:
Ah apologies. Yes, an oversight. I'm not 100% happy with the code here, it feels a little clunky passing the compression Factory around everywhere, but I can't think of a better way around it...

Reply:
Would there be a way for us (all the code in security-agent/`pkg/security`) to opt-out of this and get the previous default compression algorithm ? We don't send as many events as logs do, so we are not benefitting from advanced compression, we would be very happy is this could just be "factory.Gzip" or whatever here

Reply:
Unfortunately I don't think it would be that easy to. The compression algorithm to use is determined when the endpoints are loaded [here](https://github.com/DataDog/datadog-agent/blob/fe3db79a8ac245e8bb654ee095e9086014aa9a92/pkg/security/module/msg_sender.go#L79). Which ultimately call into the [shared log code](https://github.com/DataDog/datadog-agent/blob/fe3db79a8ac245e8bb654ee095e9086014aa9a92/comp/logs/agent/config/endpoints.go#L103).

And then the CWSReporter setsup a pipeline [here](https://github.com/DataDog/datadog-agent/blob/fe3db79a8ac245e8bb654ee095e9086014aa9a92/pkg/security/reporter/reporter.go#L56-L66) which is also in the shared log code. 

Is there a technical reason why you would want to avoid enabling `zstd` or is it about keeping the code clean?

Reply:
My main goal is to prevent the logs config to leak into CWS pipeline. It seems that this is not the case here but I'm not super sure it's just super intricated. I'm also not sure at all why this compression selection needs to be a component since the actual decision is made related to endpoints, which is happening in the logs shared code. Is the goal to move endpoints to a component as well ?

Reply:
The main reason for componentising it was because `zstd` uses cgo, which can't be used in the OTel collector. This allows OTel to use compression, but not pull in the `zstd` part. It was done like this for metrics, so I'm reusing a bunch of that work to add `zstd` to logs.

From what I can see, this will add a new config option to `runtime_security_config.endpoints.` for `compression_kind`, which can be either `gzip` or `zstd`. Currently defaulting to `gzip`, but I am planning to change it to `zstd` in time. Would that be an issue?

Reply:
I think it's all fine then